### ISSUE TITLE
Words get mashed together when being walked, if the words are around a nested list

### ISSUE THREAD

**Issue by caffodian** [PolicyStat/ckeditor-spell-check-plugin-js-dev-edge#12]
The nested list doesn't correctly separate words,

therefore 

```
            '<ul>' +
                '<li>foo' +
                    '<ol>' +
                        '<li>bar</li>' +
                    '</ol>' +
                'baz</li>' +
            '</ul>'
```

returns "foobaz" and "bar" through the wordwalker.  

https://github.com/PolicyStat/ckeditor-spell-check-plugin-js-dev-edge/blob/pstat_rewrite/tests/typowalking/wordwalker.js#L138

**Comment by caffodian** [comment]
To run this test, remove it from the ignore:

https://github.com/PolicyStat/ckeditor-spell-check-plugin-js-dev-edge/blob/pstat_rewrite/tests/typowalking/wordwalker.js#L16

**Comment by caffodian** [comment]
I think this can cause weird bugs on occasion, because it's not actually possible to create a range to mark the misdetected "foodbaz" word

**Comment by caffodian** [comment]
One way might be to have walkers end if they run into one of those nested blocks that we skip, then find a way to create a new walker that starts after the nested block?

But then I suspect there are probably very degenerate cases where someone nests multiple different lists or other skippable blocks, inside a single list item.

**Comment by caffodian** [comment]
fixed by #15
